pommerman/agents/smart_agent.py

```diff
@@ -13,8 +23,85 @@ def __init__(self, training_env=None):
         self.history = None
 
     def build_history(self, agent_obs):
-        # TODO: for Amy to finish
-        raise NotImplementedError()
+        '''Update history by incrementing timestep by 1, and
+        by new observation received from current timestep'''
+
+        def _get_history_board(board):
```

Comment:
We want to keep Fog, not clearing anything that is `not in self.history_board_values`